omnigibson/systems/macro_particle_system.py

```diff
@@ -643,13 +645,24 @@ def generate_group_particles(
         z_up[-1] = 1.0
         for position, orientation, scale, bbox_extent_local, link_prim_path in \
                 zip(positions, orientations, scales, bbox_extents_local, link_prim_paths):
+            link_name = link_prim_path.split("/")[-1]
+            link = obj.links[link_name]
             # Possibly shift the particle slightly away from the object if we're not clipping into objects
             if cls._CLIP_INTO_OBJECTS:
                 # Shift the particle halfway down
                 base_to_center = bbox_extent_local[2] / 2.0
                 normal = (T.quat2mat(orientation) @ z_up).flatten()
                 position -= normal * base_to_center
 
+            if gm.ENABLE_FLATCACHE:
+                # If flatcache is enabled, we need to modify the position we're setting because we're directly writing
+                # to the USD, which is NOT being updated when flatcache is enabled!
+                # So, we need to set the particle w.r.t. the link's initial (i.e.: spawn) pose, NOT the current pose!
+                # We will calculate the transform from the link's current pose to its initial pose, and then apply this
+                # transform to the desired global pose
+                tf = T.pose_inv(T.pose2mat(link.get_position_orientation())) @ T.pose2mat(link.spawn_position_orientation)
+                position, orientation = T.mat2pose(tf @ T.pose2mat((position, orientation)))
```

Comment:
sorry not sure what this means, maybe we can discuss in person?

Reply:
resolved!